src/transformers/models/prophetnet/modeling_prophetnet.py

```diff
@@ -171,13 +171,16 @@ def ngram_attention_bias(sequence_length, ngram, device, dtype):
     """
     This function computes the bias for the predict stream
     """
-    bias = torch.ones((ngram, sequence_length, 2 * sequence_length), device=device, dtype=dtype) * float("-inf")
+    left_block = torch.ones((ngram, sequence_length, sequence_length), device=device, dtype=dtype) * float("-inf")
+    right_block = left_block.detach().clone()
     # create bias
     for stream_idx in range(ngram):
-        for i in range(sequence_length):
-            bias[stream_idx, i, sequence_length + i] = 0
-            bias[stream_idx, i, : max(i - stream_idx, 0) + 1] = 0
-    return bias
+        right_block[stream_idx].fill_diagonal_(0, wrap=False)
+        left_block[stream_idx].triu_(-stream_idx + 1)
+        for i in range(stream_idx):
```

Comment:
could we maybe get rid of this loop as well?
Something like 

`left_block[stream_idx, torch.arange(stream_idx), 0]` should work I think

Reply:
Your new change is even better - nice :-)

Reply:
Yes, just noticed it as well. Actually pushing an even simpler version, speeding up further the calculation down to ~600us on my machine (400x faster than the baseline)